File: server/dna.core/dna.core.service/Models/ArticleModel.cs
using dna.core.data.Entities;
using dna.core.data.Infrastructure;
using dna.core.service.Models.Abstract;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace dna.core.service.Models
{

    public class ArticleModel : WriteHistoryBaseModel, IModelBase
    {
        public int Id { get; set; }
        [Required, MaxLength(100), MinLength(3)]
        public string Slug { get; set; }

        [Required, MaxLength(100), MinLength(3)]
        public string Title { get; set; }

        [Required]
        public string Description { get; set; }

        [MaxLength(255)]
        public string ShortDescription { get; set; }

        [MaxLength(255)]
        public string Metadata { get; set; }

        [Required]
        public int CategoryId { get; set; }
        public Nullable<bool> IsFeatured { get; set; }
        public Nullable<int> AcceptedById { get; set; }

        public DateTime? AcceptedDate { get; set; }
        public ArticleStatus Status { get; set; }
        
        public virtual UserModel AcceptedByUser { get; set; }
        
        public virtual ArticleCategoryModel Category { get; set; }

        public virtual IEnumerable<ArticleTagMapModel> TagMaps { get; set; }
        public virtual IEnumerable<ArticleImageModel> ImageMaps { get; set; }

        // for populate Application User ( First & Last Name }
        public virtual UserDetail Author { get; set; }

        //Is Favorite Article
        public bool IsFavorite { get; set; }
    }
}

File: server/medicore.apps/src/model/articleModel.ts
import { Type } from 'class-transformer';
import { UserModel } from './userModel';
import { ArticleCategoryModel } from './articleCategoryModel';
import { ArticleTagMapModel } from './articleTagMapModel';
import { ArticleImageModel } from './articleImageModel';
import { ArticleStatus } from './enumModel';
import { IEntityBase } from './IEntityBase';
import { WriteHistoryBaseModel } from './writeHistoryBaseModel';

export class ArticleModel extends WriteHistoryBaseModel implements IEntityBase {
    id: number;
    slug: string;
    title: string;
    description: string;
    shortDescription: string;
    metadata: string;
    categoryId: number;
    isFeatured?: boolean;
    acceptedById?: number;
    acceptedDate?: Date;
    status: ArticleStatus;
    //region model
    @Type(() => UserModel)
    acceptedByUser: UserModel;
    @Type(() => ArticleCategoryModel)
    category: ArticleCategoryModel;
    @Type(() => ArticleTagMapModel)
    tagMaps: ArticleTagMapModel[];
    @Type(() => ArticleImageModel)
    imageMaps: ArticleImageModel[];

    constructor() {
        super();
        this.id = 0;
        this.slug = null;
        this.title = null;
        this.description = null;
        this.shortDescription = null;
        this.metadata = null;
        this.categoryId = 0;
        this.isFeatured = false;
        this.acceptedById = 0;
        this.acceptedDate = null;
        this.status = null;
        this.acceptedByUser = null;
        this.category = null;
        this.tagMaps = null;
        this.imageMaps = null;
    }
}

